inmem_snapshot.go

```diff
@@ -0,0 +1,72 @@
+package raft
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+)
+
+// InmemSnapshotStore implements the SnapshotStore interface and
+// retains only the most recent snapshot
+type InmemSnapshotStore struct {
+	latest *InmemSnapshotSink
+}
+
+// InmemSnapshotSink implements SnapshotSink in memory
+type InmemSnapshotSink struct {
+	meta     SnapshotMeta
+	contents []byte
+}
+
+// NewInmemSnapshotStore creates a blank new InmemSnapshotStore
+func NewInmemSnapshotStore() *InmemSnapshotStore {
+	return &InmemSnapshotStore{
+		latest: &InmemSnapshotSink{
+			contents: []byte{},
+		},
+	}
+}
+
+// Create replaces the stored snapshot with a new one using the given args
+func (m *InmemSnapshotStore) Create(index, term uint64, peers []byte) (SnapshotSink, error) {
+	name := snapshotName(term, index)
+
+	sink := m.latest
+	sink.meta.ID = name
+	sink.meta.Index = index
+	sink.meta.Term = term
+	sink.meta.Peers = peers
+	sink.contents = []byte{}
+
+	return sink, nil
+}
+
+// List returns the latest snapshot taken
+func (m *InmemSnapshotStore) List() ([]*SnapshotMeta, error) {
+	return []*SnapshotMeta{&m.latest.meta}, nil
+}
+
+// Open wraps an io.ReadCloser around the snapshot contents
+func (m *InmemSnapshotStore) Open(id string) (*SnapshotMeta, io.ReadCloser, error) {
+	return &m.latest.meta, ioutil.NopCloser(bytes.NewReader(m.latest.contents)), nil
```

Comment:
I'd bail with an error if the ID doesn't match what's in the meta, otherwise this might cause some weird errors that will be hard to diagnose (if someone holds onto an ID and then another snapshot takes place).